broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java

```diff
@@ -643,14 +643,41 @@ private CompletableFuture<Long> popMsgFromQueue(boolean isRetry, GetMessageResul
 
                 atomicRestNum.set(result.getMaxOffset() - result.getNextBeginOffset() + atomicRestNum.get());
                 String brokerName = brokerController.getBrokerConfig().getBrokerName();
-                for (SelectMappedBufferResult mapedBuffer : result.getMessageMapedList()) {
-                    // We should not recode buffer for normal topic message
-                    if (!isRetry) {
-                        getMessageResult.addMessage(mapedBuffer);
+                for (SelectMappedBufferResult mappedBuffer : result.getMessageMapedList()) {
+                    if (MixAll.isLmq(topic)) {
+                        List<MessageExt> messageExtList = MessageDecoder.decodesBatch(mappedBuffer.getByteBuffer(),
```

Comment:
Parsing the message here will affect the performance. Could you consider compatible processing when the client parses the POP response?

Reply:
Indeed. 
I've tried to process it on the client side at first, but we can't get LMQ's queueOffset and queue ID in the client side, because the message was saved with information on the original topic and queue.

Reply:
LMQ's queueOffset is in properties of message, and queueID is default 0

Reply:
Take another case, if we hope to use the older client to visit our new broker version. Does it enough to modify in the client side?